### ISSUE TITLE
Why is num_output=2 for last deconv operation to get prediction mask

### ISSUE THREAD

**Issue by gunshi** [tinghuiz/appearance-flow#8]
Hey, was confused why num_output=2 for the shown deconv below: (from car_single/deploy.prototxt
)

```
layer {
  name: "mask_deconv15"
  type: "Deconvolution"
  bottom: "Deconvolution10"
  top: "mask_deconv15"
  convolution_param {
    num_output: 2
    pad: 1
    kernel_size: 3
    stride: 1
  }
}
```

Shouldn't this be 1 since we just want to get a binary value saying foreground (1) or background (0) for every pixel value?

Thanks!

**Comment by tinghuiz** [comment]
Here the first channel corresponds to the logit of the background class and the second corresponds to the foreground.

**Comment by gunshi** [comment]
Okay, that clears it up.
Though just using one would have been sufficient as well(thresholding score to 0 or 1)? (since an object can't be part of both foreground and background)
Thanks!

**Comment by gunshi** [comment]
Also, we won't be needing the mask predictions if we wanted to train on Kitti images right?

**Comment by tinghuiz** [comment]
That is correct as there's no 'background' pixels in KITTI.

**Comment by gunshi** [comment]
Okay.
Last question! the paper mentions that you take the 'difference' between global transformation matrices from the kitti data, but i'm guessing by difference you mean 'relative transform' and not just direct subtraction?
Just wanted a clarification on that.
Thanks

**Comment by tinghuiz** [comment]
I actually did direct subtraction, which is a bad idea in retrospect... Doing 'relative transform' should improve the performance.

**Comment by gunshi** [comment]
Okay, thanks!
